llvm/lib/Target/AIE/AIEPostPipeliner.h

```diff
@@ -43,14 +44,65 @@ class NodeInfo {
   int ModuloCycle = 0;
   // Cycle / II
   int Stage = 0;
+
+  int StaticEarliest = 0;
+  int StaticLatest = -1;
+
   // The earliest cycle at which this can be scheduled to meet latencies
   // This includes the lowerbound of the modulo condition, i.e.
   // Earliest(N) >= Cycle(N - NInstr) + II
   int Earliest = 0;
-  // For an LCD K1 -> K2, this holds II + Earliest(K2 - NInstr) - Latency(LCD)
-  // Instructions with lower Latest have higher priority in the
-  // top down scheduling
-  int Latest = 0;
+
+  // The latest cycle at which this can be scheduled. This is a negative value
+  // relative to the length of the linear schedule.
```

Comment:
Nit: Add a note to say that `-1` represents the last cycle of the schedule, while `-Length` represents the first one.